js/ui/discoveryFeedButton.js

```diff
@@ -51,12 +51,10 @@ const DiscoveryFeedButton = new Lang.Class({
     Extends: St.Button,
 
     _init: function() {
-        let iconFile = Gio.File.new_for_uri('resource:///org/gnome/shell/theme/discovery-feed-open-tab.png');
-        let gicon = new Gio.FileIcon({ file: iconFile });
-        this._icon = new St.Icon({ gicon: gicon,
-                                   style_class: 'discovery-feed-icon',
-                                   track_hover: true });
+        this._icon = new St.Icon({style_class: 'discovery-feed-tile-icon',
```

Comment:
Event better, thanks for the suggestion - it really starts to look nice clean code now.